src/h5web/vis-packs/core/shared/AxisSystemProvider.tsx

```diff
@@ -0,0 +1,37 @@
+import type { ReactNode } from 'react';
+import type { AxisConfig } from '../models';
+import { useVisSize } from '../hooks';
+import { AxisSystemContext } from './AxisSystemContext';
+import { getCanvasScale } from '../utils';
+
+interface Props {
+  visRatio?: number;
+  abscissaConfig: AxisConfig;
+  ordinateConfig: AxisConfig;
+  children: ReactNode;
+}
+
+function AxisSystemProvider(props: Props) {
+  const { visRatio, abscissaConfig, ordinateConfig, children } = props;
+
+  const visSize = useVisSize(visRatio);
+  const abscissaScale = getCanvasScale(abscissaConfig, visSize.width);
+  const ordinateScale = getCanvasScale(ordinateConfig, visSize.height);
```

Comment:
When the provider renders, it causes `PanZoomMesh` to render, which causes R3F's `invalidate`. to be called, which renders a new frame. Previously, since we had `useFrameRendering` inside the provider, it meant that the provider would re-render on every frame, which led to an infinite render loop.

Now only the child `AxisSystem` component re-renders on every frame. The provider now only renders when the size of the canvas changes.